x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleService.java

```diff
@@ -88,9 +91,11 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
 
         boolean isWatcherStoppedManually = isWatcherStoppedManually(event.state());
+        boolean isStoppedOrStopping = this.state.get() == WatcherState.STOPPED ||
+            this.state.get() == WatcherState.STOPPING;
         // if this is not a data node, we need to start it ourselves possibly
         if (event.state().nodes().getLocalNode().isDataNode() == false &&
-            isWatcherStoppedManually == false && this.state.get() == WatcherState.STOPPED) {
+            isWatcherStoppedManually == false && isStoppedOrStopping) {
             this.state.set(WatcherState.STARTING);
```

Comment:
Shouldn't we only be able to set the state to `STARTING` if the current state is `STOPPED`?

Reply:
This is to stay passive with the old behavior.  If we kept this to only `STOPPED` with this change, it would mean you can not restart watcher while any watches are currently inflight. 

The pre-existing design allows you "stop" and restart it immediately and re-process the same watch, any inflight watches will finish up. If I didn't allow `STOPPING` here you would have to wait for all inflight watches to finish up and if one were were stuck (for what ever reason) could cause the in-ability to ever reach a fully `STOPPED` state.